00 - Contribute to the Obsidian Hub/03 Contributor Notes/03.02 Design Decisions/hub.yaml proposal for themes.md

```diff
@@ -0,0 +1,303 @@
+          custom_callouts:
+            docs:
+               file: docs/dnd_callouts_overview.md # from root of repo, to be included as is
+   ```
+- A link to your own documentation
+  ```yaml
+  theme:
+    features: 
+          custom_callouts:
+            docs:
+		       links: # List of strings (can be markdown-formatted)
+		           - [More information](www.mytheme.com/dnd_callouts)
+   ```
+- Combinations of the above options (except boolean)
+  ```yaml
+  theme:
+    features: 
+          custom_callouts:
+            docs:
+               file: docs/dnd_callouts_overview.md # from root of repo, to be included as is
+		       links: # List of strings (can be markdown-formatted)
+		           - [More information](www.mytheme.com/dnd_callouts)
+            screenshots: # Path to screenshots for this feature (from the root of the repo)
+                - file: docs/all_dnd_callouts.png
+                  caption: All the D&D callouts for My Theme
+                - file: docs/treasure_dnd_callout.png
+                  caption: Example of the treasure D&D callout
+                  
+   ```
+
+%% Note: for the implementation we'd need to know how to structure the templates so that combinations of these make sense %%
+
+### Plugin support
+
+This key (`plugin_support`) is essentially an alternative to parsing the [compatible plugins](https://publish.obsidian.md/hub/04+-+Guides%2C+Workflows%2C+%26+Courses/Guides/Adding+plugin+compatibility+for+themes+to+the+Obsidian+Hub) from the `obsidian.css` file if you prefer to have the hub stuff outside of your theme file. If you don't specify this key, we'll still try to read your plugins from the css file.
+
+```yaml
+theme:
+    plugin_support: [] # IDs of the plugins that have been styled by this theme (adding them here would ignore the plugin IDs on the obsidian.css file)
+```
+
+### Screenshots
+
+If you want to add additional screenshots, or don't want to add features but prefer to just add screenshots of your theme, you can use the `screenshot` key:
+
+```yaml
+theme:
+    screenshots: # Path to other screenshots from the root of the repo
+        - file: 
+          caption:
+```
+
+## Template
+
+Here is a full template with all the options being proposed:
+
+```yaml
+docs: # Link to the theme documentation
+hub_content: [], # List of markdown documents to be included as is to the hub relative to the root of the repo.
+authors: # More than one author is possible, start a new line with - 
+    - github: # GitHub user names of the people who created the plugin or theme
+      websites: [] # Links to personal websites
+      discord:
+      twitter:
+      youtube:
+      sponsorships:
+          buymeacoffee: # Add username to kofi or buy me a coffee here
+          paypal: 
+          github:
+          patreon:
+          others: [] # List of formatted markdown strings
+maintainers: [] # GitHub user names of those maintaining the plugin (leave empty if it's the same as author)
```

Comment:
usually, maintainers are also theme devs. not sure whether this is needed?

Reply:
Yeah, might not be a common practice in themes, but it's something still available (because of plugins). I could remove it from the docs if it makes it confusing.